diff --git a/node_modules/bitcore-wallet-client/src/lib/key.ts b/node_modules/bitcore-wallet-client/src/lib/key.ts
index 3769b38..9c56961 100644
--- a/node_modules/bitcore-wallet-client/src/lib/key.ts
+++ b/node_modules/bitcore-wallet-client/src/lib/key.ts
@@ -8,7 +8,6 @@ import {
   Transactions
 } from 'crypto-wallet-core';
 import * as _ from 'lodash';
-import 'source-map-support/register';
 import { Constants, Utils } from './common';
 import { Credentials } from './credentials';
 
diff --git a/node_modules/bitcore-wallet-client/ts_build/lib/key.js b/node_modules/bitcore-wallet-client/ts_build/lib/key.js
index 2bb5038..a56c693 100644
--- a/node_modules/bitcore-wallet-client/ts_build/lib/key.js
+++ b/node_modules/bitcore-wallet-client/ts_build/lib/key.js
@@ -37,7 +37,6 @@ exports.Key = void 0;
 var $ = require('preconditions').singleton();
 const crypto_wallet_core_1 = require("crypto-wallet-core");
 const _ = __importStar(require("lodash"));
-require("source-map-support/register");
 const common_1 = require("./common");
 const credentials_1 = require("./credentials");
 var Bitcore = crypto_wallet_core_1.BitcoreLib;
