diff --git a/node_modules/crypto-wallet-core/ts_build/src/validation/xrp/index.js b/node_modules/crypto-wallet-core/ts_build/src/validation/xrp/index.js
index 59f4b97..fe9df8f 100644
--- a/node_modules/crypto-wallet-core/ts_build/src/validation/xrp/index.js
+++ b/node_modules/crypto-wallet-core/ts_build/src/validation/xrp/index.js
@@ -17,7 +17,7 @@ class XrpValidation {
         const buffer = Buffer.from(base58.decode(address));
         const prefix = buffer.subarray(0, 1);
         const data = buffer.subarray(1, -4);
-        let hash = Buffer.concat([prefix, data]);
+        let hash = Buffer.concat([Buffer.from(prefix), Buffer.from(data)]);
         hash = bitcore_lib_1.default.crypto.Hash.sha256(hash);
         hash = bitcore_lib_1.default.crypto.Hash.sha256(hash);
         const checksum = buffer.subarray(-4).reduce((acc, check, index) => {
