diff --git a/node_modules/react-native-webassembly/android/CMakeLists.txt b/node_modules/react-native-webassembly/android/CMakeLists.txt
index 0f879e0..75b0cdc 100644
--- a/node_modules/react-native-webassembly/android/CMakeLists.txt
+++ b/node_modules/react-native-webassembly/android/CMakeLists.txt
@@ -64,7 +64,9 @@ find_package(ReactAndroid REQUIRED CONFIG)
 
 target_link_libraries(
         cpp
+        "-Wl,-z,max-page-size=16384"
         ${log-lib}
+        c++_shared
         ReactAndroid::jsi
         android
 )
diff --git a/node_modules/react-native-webassembly/cpp/m3_config_platforms.h b/node_modules/react-native-webassembly/cpp/m3_config_platforms.h
index 50b86ac..8deaf8d 100644
--- a/node_modules/react-native-webassembly/cpp/m3_config_platforms.h
+++ b/node_modules/react-native-webassembly/cpp/m3_config_platforms.h
@@ -82,7 +82,7 @@
 #  endif
 # endif
 
-# if M3_COMPILER_HAS_ATTRIBUTE(musttail)
+# if M3_COMPILER_HAS_ATTRIBUTE(musttail) && !defined(__arm__)
 #   define M3_MUSTTAIL __attribute__((musttail))
 # else
 #   define M3_MUSTTAIL
diff --git a/node_modules/react-native-webassembly/ios/Webassembly.mm b/node_modules/react-native-webassembly/ios/Webassembly.mm
index 58d2b84..64991b4 100644
--- a/node_modules/react-native-webassembly/ios/Webassembly.mm
+++ b/node_modules/react-native-webassembly/ios/Webassembly.mm
@@ -8,0 +9,5 @@
+#ifdef RCT_NEW_ARCH_ENABLED
+#import <ReactCommon/RCTInteropTurboModule.h>
+#import <ReactCommon/RCTTurboModuleWithJSIBindings.h>
+#endif
+
@@ -11,0 +17,6 @@
+#ifdef RCT_NEW_ARCH_ENABLED
+@interface Webassembly () <RCTTurboModule, RCTTurboModuleWithJSIBindings>
+@property (nonatomic, assign) BOOL didInstallJSIBindings;
+@end
+#endif
+
@@ -25 +36,4 @@ + (BOOL)requiresMainQueueSetup {
-  RCTBridge* bridge = [RCTBridge currentBridge];
+#ifdef RCT_NEW_ARCH_ENABLED
+  return @(self.didInstallJSIBindings);
+#else
+  RCTBridge* bridge = self.bridge ?: [RCTBridge currentBridge];
@@ -37,0 +51 @@ + (BOOL)requiresMainQueueSetup {
+#endif
@@ -38,0 +54,15 @@ + (BOOL)requiresMainQueueSetup {
+#ifdef RCT_NEW_ARCH_ENABLED
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
+    (const facebook::react::ObjCTurboModule::InitParams &)params
+{
+  return std::make_shared<facebook::react::ObjCInteropTurboModule>(params);
+}
+
+- (void)installJSIBindingsWithRuntime:(facebook::jsi::Runtime &)runtime
+                          callInvoker:(const std::shared_ptr<facebook::react::CallInvoker> &)callInvoker
+{
+  webassembly::install(runtime);
+  self.didInstallJSIBindings = YES;
+}
+#endif
+
diff --git a/node_modules/react-native-webassembly/react-native-webassembly.podspec b/node_modules/react-native-webassembly/react-native-webassembly.podspec
index aba1f2f..747b99e 100644
--- a/node_modules/react-native-webassembly/react-native-webassembly.podspec
+++ b/node_modules/react-native-webassembly/react-native-webassembly.podspec
@@ -4 +3,0 @@ package = JSON.parse(File.read(File.join(__dir__, "package.json")))
-folly_compiler_flags = '-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1 -Wno-comma -Wno-shorten-64-to-32'
@@ -22,2 +20,0 @@ Pod::Spec.new do |s|
-  s.dependency "React-Core"
-
@@ -31,6 +27 @@ Pod::Spec.new do |s|
-  s.compiler_flags = folly_compiler_flags + " -DRCT_NEW_ARCH_ENABLED=1"
-  s.dependency "React-Codegen"
-  s.dependency "RCT-Folly"
-  s.dependency "RCTRequired"
-  s.dependency "RCTTypeSafety"
-  s.dependency "ReactCommon/turbomodule/core"
+  install_modules_dependencies(s)
