diff --git a/node_modules/react-native-appsflyer/android/build.gradle b/node_modules/react-native-appsflyer/android/build.gradle
index b767abe..8e6fcc2 100755
--- a/node_modules/react-native-appsflyer/android/build.gradle
+++ b/node_modules/react-native-appsflyer/android/build.gradle
@@ -28,3 +28,7 @@ android {
     buildToolsVersion safeExtGet('buildToolsVersion', '34.0.0')
-
+
+    buildFeatures {
+        buildConfig true
+    }
+
     defaultConfig {
