diff --git a/node_modules/react-native-reanimated/Common/cpp/SharedItems/Shareables.h b/node_modules/react-native-reanimated/Common/cpp/SharedItems/Shareables.h
index f94ec73..b7f4840 100644
--- a/node_modules/react-native-reanimated/Common/cpp/SharedItems/Shareables.h
+++ b/node_modules/react-native-reanimated/Common/cpp/SharedItems/Shareables.h
@@ -236,7 +236,7 @@ class ShareableArrayBuffer : public Shareable {
   ShareableArrayBuffer(
       jsi::Runtime &rt,
 #if REACT_NATIVE_MINOR_VERSION >= 72
-      const jsi::ArrayBuffer &arrayBuffer
+      jsi::ArrayBuffer arrayBuffer
 #else
       jsi::ArrayBuffer arrayBuffer
 #endif
