diff --git a/node_modules/react-native-modal/dist/modal.d.ts b/node_modules/react-native-modal/dist/modal.d.ts
index 27596b9..7646b92 100644
--- a/node_modules/react-native-modal/dist/modal.d.ts
+++ b/node_modules/react-native-modal/dist/modal.d.ts
@@ -108,7 +108,6 @@ export declare class ReactNativeModal extends React.Component<ModalProps, State>
     contentRef: any;
     panResponder: OrNull<PanResponderInstance>;
     didUpdateDimensionsEmitter: OrNull<EmitterSubscription>;
-    interactionHandle: OrNull<number>;
     constructor(props: ModalProps);
     static getDerivedStateFromProps(nextProps: Readonly<ModalProps>, state: State): {
         isVisible: boolean;
diff --git a/node_modules/react-native-modal/dist/modal.js b/node_modules/react-native-modal/dist/modal.js
index 74edee4..e1a9f18 100644
--- a/node_modules/react-native-modal/dist/modal.js
+++ b/node_modules/react-native-modal/dist/modal.js
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import { Animated, DeviceEventEmitter, Dimensions, InteractionManager, KeyboardAvoidingView, Modal, PanResponder, Platform, TouchableWithoutFeedback, View, } from 'react-native';
+import { Animated, DeviceEventEmitter, Dimensions, KeyboardAvoidingView, Modal, PanResponder, Platform, TouchableWithoutFeedback, View, } from 'react-native';
 import * as animatable from 'react-native-animatable';
 import { buildAnimations, initializeAnimations, reversePercentage, } from './utils';
 import styles from './modal.style';
@@ -69,7 +69,6 @@ export class ReactNativeModal extends React.Component {
     contentRef;
     panResponder = null;
     didUpdateDimensionsEmitter = null;
-    interactionHandle = null;
     constructor(props) {
         super(props);
         const { animationIn, animationOut } = buildAnimations(extractAnimationFromProps(props));
@@ -115,10 +114,6 @@ export class ReactNativeModal extends React.Component {
         if (this.didUpdateDimensionsEmitter) {
             this.didUpdateDimensionsEmitter.remove();
         }
-        if (this.interactionHandle) {
-            InteractionManager.clearInteractionHandle(this.interactionHandle);
-            this.interactionHandle = null;
-        }
     }
     componentDidUpdate(prevProps) {
         // If the animations have been changed then rebuild them to make sure we're
@@ -393,17 +388,10 @@ export class ReactNativeModal extends React.Component {
         }
         if (this.contentRef) {
             this.props.onModalWillShow && this.props.onModalWillShow();
-            if (this.interactionHandle == null) {
-                this.interactionHandle = InteractionManager.createInteractionHandle();
-            }
             this.contentRef
                 .animate(this.animationIn, this.props.animationInTiming)
                 .then(() => {
                 this.isTransitioning = false;
-                if (this.interactionHandle) {
-                    InteractionManager.clearInteractionHandle(this.interactionHandle);
-                    this.interactionHandle = null;
-                }
                 if (!this.props.isVisible) {
                     this.close();
                 }
@@ -439,17 +427,10 @@ export class ReactNativeModal extends React.Component {
         }
         if (this.contentRef) {
             this.props.onModalWillHide && this.props.onModalWillHide();
-            if (this.interactionHandle == null) {
-                this.interactionHandle = InteractionManager.createInteractionHandle();
-            }
             this.contentRef
                 .animate(animationOut, this.props.animationOutTiming)
                 .then(() => {
                 this.isTransitioning = false;
-                if (this.interactionHandle) {
-                    InteractionManager.clearInteractionHandle(this.interactionHandle);
-                    this.interactionHandle = null;
-                }
                 if (this.props.isVisible) {
                     this.open();
                 }
